src/charm.py

```diff
@@ -205,6 +225,26 @@ def __init__(self, *args, **kargs) -> None:
         )
         self.framework.observe(self.on.update_status, self._on_update_status)
 
+    def setup_state(self) -> None:
+        """Set up the charm state."""
+        try:
+            self.state = State.from_charm(self)
+        except CharmConfigInvalidError as exc:
+            self.unit.status = ops.BlockedStatus(exc.msg)
+            raise ConfigurationError(exc.msg) from exc
```

Comment:
I'm wondering if `BlockedStatus` would be overridden by `ConfigurationError` (`ErrorStatus`).

Reply:
The `ConfigurationError` would be caught by `catch_charm_errors` declarator and be in `BlockedStatus`.
The line 233 is a bit unnecessary. I have removed that.